src/simd/mod.rs

```diff
@@ -1,14 +1,41 @@
 use crate::util;
 
+/// Fallback functions, used for search_non_ident on unsupported architectures
+pub mod fallback;
+
+// Architecture-specific SIMD implementations
+#[cfg(target_arch = "x86_64")]
+mod x86_64;
+
+#[cfg(target_arch = "aarch64")]
+mod aarch64;
+
 /// Checks if the given byte is a "closing" byte (/ or >)
 #[inline]
 pub fn is_closing(needle: u8) -> bool {
     (needle == b'/') | (needle == b'>')
 }
 
 /// Searches for the first non-identifier in `haystack`
+#[inline]
 pub fn search_non_ident(haystack: &[u8]) -> Option<usize> {
-    haystack.iter().position(|&c| !util::is_ident(c))
+    #[cfg(target_arch = "x86_64")]
```

Comment:
```suggestion
    #[cfg(all(target_arch = "x86_64", target_feature = "sse2"))]
```

Basically, [it is technically possible to compile for `x86-64` but without SSE2 support](https://github.com/BurntSushi/memchr/pull/77). In particular, SSE2 requires `xmm` registers, and that in turn requires cooperation with the kernel to support them.